src/octoprint/plugins/softwareupdate/__init__.py

```diff
@@ -1492,6 +1552,34 @@ def get_wizard_details(self):
     def on_event(self, event, payload):
         from octoprint.events import Events
 
+        if event == Events.PRINT_STARTED:
+            self._queued_updates_timer_stop()
+
+        if (
+            event == Events.PRINT_DONE
+            and self._settings.global_get(["webcam", "timelapse", "type"]) == "off"
+            and len(self._queued_updates.get("targets", [])) > 0
+        ):
+            self._queued_updates_timer_start()
+
+        if (
+            event == Events.MOVIE_DONE
+            and self._settings.global_get(["webcam", "timelapse", "type"]) != "off"
+            and len(self._queued_updates.get("targets", [])) > 0
+            and not (self._printer.is_printing() or self._printer.is_paused())
+        ):
+            self._queued_updates_timer_start()
+
+        if (
+            event == Events.PRINT_FAILED
+            and self._settings.global_get(["webcam", "timelapse", "type"]) == "off"
+            and len(self._queued_updates.get("targets", [])) > 0
+        ):
+            self._send_client_message(
+                "queued_updates",
+                {"print_failed": True, "targets": self._queued_updates["targets"]},
+            )
+
```

Comment:
Should be `if ... elif ...` given that there's no way more than one of these can ever be true.